### ISSUE TITLE
[Improvement] Rename default `master` branch to `main`

### ISSUE THREAD

**Issue by arturcic** [GitTools/GitVersion#2570]
GitHub already added the feature to rename the default branch and re-target the PRs on the new default branch.

@GitTools/developers  do you agree with that?

It will require also adapting the build scripts and maybe the code itself

**Comment by asbjornu** [comment]
Yes, 👍🏼 from me.

**Comment by gep13** [comment]
Seems like a good idea to me!

**Comment by dazinator** [comment]
Whats the benefit?

**Comment by asbjornu** [comment]
There's a lot of good information [here](https://github.com/github/renaming), @dazinator.

**Comment by dazinator** [comment]
Oh dear. Politics. 
I guess everyone who works in [master data management](https://en.wikipedia.org/wiki/Master_data_management) is hoping they will go under the radar of this particular political agenda!

**Comment by GeertvanHorrik** [comment]
@arturcic is this a "visual" / GitVersion branch change only or also a *behavioral runtime* change in how GitVersion works?

**Comment by arturcic** [comment]
@GeertvanHorrik the tests internally use `main` to make it consistent, but `master` is still supported. I introduced the constant `MainBranch` that is used in the tests, and before the switch it had the `master` value. This is for the internal code only. As for the consumer, default is `main`, but you can also use the master in `GitVersion.yml` as there is a compatibility added.

**Comment by arturcic** [comment]
That means projects that were using the GitVersion before this change will continue to work

**Comment by GeertvanHorrik** [comment]
Thanks for the explanation, then I don't mind, as long as it's not a breaking change for people sticking with master  👍

**Comment by gep13** [comment]
This is a nice touch...

![image](https://user-images.githubusercontent.com/1271146/106509336-6887d780-64c5-11eb-86c7-9d4bf3b237f0.png)

Got that when browsing to repository after rename.

**Comment by david-driscoll** [comment]
Just an FYI I found that this causes a failure when using a branch configuration that currently references `master`.  Changing the `source-branches` fixes the problem however that seems like an accidental breaking change.  ❤️ 

```yml
assembly-versioning-scheme: MajorMinorPatch
mode: ContinuousDeployment
continuous-delivery-fallback-tag: ci
next-version: 0.1.0
branches:
  next:
    regex: ^next$
    mode: ContinuousDeployment
    tag: next
    increment: Major
    prevent-increment-of-merged-branch-version: false
    track-merge-target: false
    tracks-release-branches: false
    is-release-branch: false
    source-branches: ['master']
ignore:
  sha: []
```

**Comment by asbjornu** [comment]
@david-driscoll, this should have been fixed in #2588.
